src/main/java/net/smartcosmos/dao/things/repository/ThingRepository.java

```diff
@@ -17,17 +17,17 @@
 public interface ThingRepository extends JpaRepository<ThingEntity, UUID>, PagingAndSortingRepository<ThingEntity, UUID>, QueryByExampleExecutor<ThingEntity>, JpaSpecificationExecutor<ThingEntity>
 {
     @Transactional
-    List<ThingEntity> deleteByIdAndTenantIdAndType(UUID id, UUID tenantId, String type);
+    List<ThingEntity> deleteByIdAndTenantIdAndTypeIgnoreCase(UUID id, UUID tenantId, String type);
```

Comment:
Unless there is a signature overlap I do not see it is not necessary to have such long method names. You should be able to go with
`findby(UUID thingId, UUID tenantId, String type)` and
`findbyIgnoreCase(UUID thingId, UUID tenantId, String type)`